src/features/expressions/useExpressions.ts

```diff
@@ -52,17 +58,21 @@ export function useExpressions<T>(input: T, _options?: UseExpressionsOptions<T>)
   const instanceContextSelector = getInstanceContextSelector();
   const instanceContext: IInstanceContext = useAppSelector(instanceContextSelector);
   const id = (options && options.forComponentId) || component.id;
+  const rowIndex: number | undefined = options && options.rowIndex;
 
   const node = useMemo(() => {
     if (id) {
       const foundNode = nodes.findById(id);
+      if (typeof rowIndex == 'number' && foundNode && foundNode.item.type == 'Group') {
+        return foundNode.children(undefined, rowIndex)[0];
+      }
```

Comment:
Glorious!! :star_struck: Simple, easy and efficient. This is much better than the solution I was thinking of.